test/test_selectors.py

```diff
@@ -613,6 +541,111 @@ def test_leaking_fds(self):
         after_fds = len(proc.open_files())
         self.assertEqual(before_fds, after_fds)
 
+    @skipIf(sys.platform == "win32", "psutil.Process.open_files() is unstable on Windows.")
+    def test_wait_read_leaking_fds(self):
+        self.patch_wait_selector()
+        proc = psutil.Process()
+        rd, wr = self.make_socketpair()
+        before_fds = len(proc.open_files())
+        self.assertEqual(0, len(wait.wait_for_read([rd], 0.001)))
+        after_fds = len(proc.open_files())
+        self.assertEqual(before_fds, after_fds)
```

Comment:
There is a test to check whether close was called below this one, I'll remove the fd tests in favor of two tests (`wait_for_read` and `wait_for_write`) for if the selector is properly closed. :)